pkg/config/dynamic/http_config.go

```diff
@@ -208,6 +210,8 @@ type ServersTransport struct {
 	Certificates        tls.Certificates    `description:"Certificates for mTLS." json:"certificates,omitempty" toml:"certificates,omitempty" yaml:"certificates,omitempty" export:"true"`
 	MaxIdleConnsPerHost int                 `description:"If non-zero, controls the maximum idle (keep-alive) to keep per-host. If zero, DefaultMaxIdleConnsPerHost is used" json:"maxIdleConnsPerHost,omitempty" toml:"maxIdleConnsPerHost,omitempty" yaml:"maxIdleConnsPerHost,omitempty" export:"true"`
 	ForwardingTimeouts  *ForwardingTimeouts `description:"Timeouts for requests forwarded to the backend servers." json:"forwardingTimeouts,omitempty" toml:"forwardingTimeouts,omitempty" yaml:"forwardingTimeouts,omitempty" export:"true"`
+	// TODO: Verify with traefik team if there is a better way to expose this
+	VerifyPeerCertificate func(cfg *gtls.Config, rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error `json:"-" hash:"-"`
```

Comment:
I'm trying to see if we can make this all work just for Consul Connect without adding that exported field. Not obvious to me for now though.

Reply:
Also, why does it not have the same tags as the other fields (and especially no omitempty)?

Reply:
Because I only added what was needed to get the tests passing for now. But yes those might need cleanup. I am not sure that `omitEmpty` makes much sense though, given that "-" prevents exporting anyways. Do you happen to know where the `export` tag is coming from?

Reply:
apparently it's supposed to be set for the fields that we are interested in being sent to Pilot, when the user has agreed to send their (anonymised) data. Given that we don't want this field to be serializable for now, we probably want to let export be false for now.
However, I think we should already set the omitEmpty for now, just in case. Especially so that it's already set properly if we ever decide to expose that field.
Also, I think we should we add toml:"-" as well. Maybe it's not needed, but since Server.Scheme has all three of json, toml, yaml ,  I think we should do the same here.